src/parser/Expression.ts

```diff
@@ -645,5 +645,112 @@ export class CallfuncExpression implements Expression {
         );
         return result;
     }
+}
+
+export class ConditionalExpression implements Expression {
+    constructor(
+        readonly test: Expression,
+        readonly consequent: Expression,
+        readonly alternate: Expression
+    ) {
+        this.range = Range.create(
+            test.range.start,
+            alternate.range.end
+        );
+    }
 
+    public readonly range: Range;
+
+    transpile(state: TranspileState) {
+        let result = [];
+        let [testExpressions, testVarExpressions, testUniqueNames] = this.getExpressionInfo(this.test);
+        let [consequentExpressions, consequentVarExpressions, consequentUniqueNames] = this.getExpressionInfo(this.consequent);
+        let [alternateExpressions, alternateVarExpressions, alternateUniqueNames] = this.getExpressionInfo(this.alternate);
+
+        let allExpressions = [...testExpressions, ...consequentExpressions, ...alternateExpressions];
+        let allUniqueVarNames = [...new Set([...testUniqueNames, ...consequentUniqueNames, ...alternateUniqueNames])];
+
+        let mutatingExpressions = allExpressions.filter(e => e instanceof CallExpression || e instanceof CallfuncExpression || e instanceof DottedGetExpression);
+
+        // - TODO - it doesn't look like we need to manipulate the variable names
+        // we can assign m on scope, and it should be fine
+
+        //FIXME - need to ensure I'm using SourceNode's correctly here - not sure how to do that yet
+        if (mutatingExpressions.length > 0) {
+            //we need to do a scope-safe ternary operation
+            let scope = '{';
+            // eslint-disable-next-line no-return-assign
+            allUniqueVarNames.forEach(name => scope += `\n"${name}": name`);
+            scope += '}';
+
+            result.push(`bslib_scopeSafeTernary(${this.test.transpile(state)}, ${scope},`);
+            result = result.concat(this.getScopedFunction(state, this.consequent, consequentUniqueNames));
+            result = result.concat(this.getScopedFunction(state, this.alternate, alternateUniqueNames));
+            result.push(')');
+        } else {
+            const ternaryCall = `bslib_simpleTernary(${this.test.transpile(state)}, ${this.consequent.transpile(state)}, ${this.alternate.transpile((state))})`;
```

Comment:
This isn't right. transpile returns `Array<string|SourceNode` so this would end up calling `array.toString()`. You need to break these out.

Something to the effect of
```
result.push('bslib_simpleTernary(');
result.push(...this.test.transpile(state));
result.push(', ');
result.push(...this.consequent.transpile(state));
result.push(', ');
result.push(...this.alternate.transpile(state));
result.push(')');
```

Reply:
yep - I realized this in my ternary-null coalesce pr and fixed it there. Will fix it here too.